logging/MonologFactory.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace SilverStripe\Framework\Logging;
+
+/**
+ * Adaptor connecting Monolog to SilverStripe for error handling
+ */
+class MonologFactory extends \InjectionCreator
+{
+	public function create($class, array $params = array()) {
+		// Register the default error handler
+		$logger = parent::create($class, $params);
+
+		// Load the error handlers into the monolog loader
+		$injector = \Injector::inst();
+
+		$handlerID = null;
+		$handlers = \Config::inst()->get('Logger', 'handlers');
+		foreach($handlers as $handler) {
+			// A hacky mechanism for disabling handlers in yaml config
+			if(in_array(strtolower($handler['class']), array('none', 'disable'))) {
+				continue;
+			}
+
+			// Special treatment for Formatter
+			if(isset($handler['properties']['Formatter']['class'])) {
```

Comment:
>Why not just make the handlers a dependency of MonologFactory itself.

My main concern with that is that it makes it quite arbitrary where you are supposed to insert the configuration. You need to be aware of the MonologFactory and how it works. It starts getting quite verbose, as well.

    Injector:
      SilverStripe\Framework\Logging\MonologFactory:
        properties:
          handlers:
            Mail:
              class: Monolog\Handler\NativeMailerHandler
              ...

Maybe it would be better if it were injected into the Logger instead?

    Injector:
      Logger:
        properties:
          handlers:
            Mail:
              class: Monolog\Handler\NativeMailerHandler
              ...

It would be good to get a broader set of views as to the direction we should go on this. It's pretty straightforward to implement whichever approach we want; it's a trade-off between configuration brevity and minimising the amount of custom instantiation code we have.

My preference would be to go with an approach that we can consistently use and is reasonably enjoyable and clear for developers to use—I don't want people to have to learn "programming in Injector".

> I don't see why you can't just use Config api if you need a finer degree of control in your app

I don't really know what you mean by this; it's more helpful to give specific examples. The use-case that I had in mind (disabling the default handler) wasn't well catered for by the config API. Moving from an array to a map helps (because you can name the items that you want to disable) but there's a change to the config API necessary to make it work without my `if class == "none" or "disabled"` check. It sounds like this would be a reasonable ss4 change.